### PR TITLE
Fetch root agent id before executing the agent

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description
Fetch root agent id before executing the agent every time, and remove some unused unit test code.

### Issues Resolved
https://github.com/opensearch-project/dashboards-assistant/issues/154

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test.
- [x] New functionality has user manual doc added.
- [x] Commits are signed per the DCO using --signoff.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by lezzago** [comment]
Why does this change need to happen? Was this to deal with scenarios when the agent id changes? If so, can we try to refetch the root agent if the call fails to the execute agent api with the old root agent id?

File: server/services/chat/olly_chat_service.ts [hunk 1]
```diff
@@ -58,40 +48,22 @@ export class OllyChatService implements ChatService {
...
-        const oldRootAgentId = OllyChatService.rootAgentId;
-        await this.initRootAgent();
-        this.context.assistant_plugin.logger.info(
-          `cannot find the root agent id: ${oldRootAgentId}, try to fetch the new root agent id, now it is:${OllyChatService.rootAgentId}`
-        );
-        return await this.callExecuteAgentAPI(payload);
-      } else {
-        throw error;
-      }
+      throw error;
```
Comment (joshuali925) [inline server/services/chat/olly_chat_service.ts:56]: since it's just `throw error`, maybe remove the try catch?

**Comment by joshuali925** [comment]
> Why does this change need to happen? Was this to deal with scenarios when the agent id changes? If so, can we try to refetch the root agent if the call fails to the execute agent api with the old root agent id?

@lezzago it's explained in the linked issue, if we cache the agent id, user won't be able to change agent unless they delete the old one. cache was added to save API calls, but the get config API has lower overhead

**Comment by lezzago** [comment]
> But i still think it's a rare operation to change agent ID. although it's not the bottleneck and get call is lightweight, it is still some overhead to every user request

Ah ok, that makes and I agree, the get config api is a lightweight call. We may need to revisit this if we start to see all these overhead calls adds problems, especially if the dashboard assistant apis are getting by outside services and not being used just on OpenSearch Dashboards.

**Comment by xluo-aws** [comment]
> > But i still think it's a rare operation to change agent ID. although it's not the bottleneck and get call is lightweight, it is still some overhead to every user request
> 
> Ah ok, that makes and I agree, the get config api is a lightweight call. We may need to revisit this if we start to see all these overhead calls adds problems, especially if the dashboard assistant apis are getting by outside services and not being used just on OpenSearch Dashboards.

Yes, we can start with this approach. One future optimization is to add cache but it introduces complexity so we don't want to add it until we see this API adds more and more overhead.

File: server/services/chat/olly_chat_service.ts (hunk 1 above)
Comment (gaobinlong) [reply server/services/chat/olly_chat_service.ts]: Yeah, I've changed that, thanks!

**Review by lezzago — APPROVED** [review]

**Pull request merged**